src/Data.js

```diff
@@ -40,9 +40,9 @@ export default function Data({ shx }) {
   
   const renderNeedPasscode = () => {
 
-	const msg = (passcode
-				 ? "Given passcode not valid for this SMART Health Link."
-				 : "This SMART Health Link requires a passcode.");
+    const msg = (passcode
+                ? shxResult.reasons[0]  // Display the specific error message from PasscodeError
```

Comment:
leaving a few 'defensive' coding suggestions since the SHL spec is so new and I'm betting many folks will have different implementations of these errors. Here I'd test for something like (Array.isArray(shxResult.reasons) && shlResult.reasons.length > 0) before doing the dereference and having a fallback generic error.